src/ZGPeerSession.cpp

```diff
@@ -262,7 +267,7 @@ void ZGPeerSession :: MessageReceivedFromPeer(const ZGPeerID & fromPeerID, const
       case PZG_PEER_COMMAND_USER_MESSAGE:
       {
          MessageRef userMsg = msg()->GetMessage(PZG_PEER_NAME_USER_MESSAGE);
-         if (userMsg()) MessageReceivedFromPeer(fromPeerID, userMsg);
+         if (userMsg()) PrivateMessageReceivedFromPeer(fromPeerID, userMsg);
```

Comment:
^ I think this line should call MessageReceivedFromPeer() rather than PrivateMessageReceivedFromPeer()?  Because the Message contained in the PZG_PEER_NAME_USER_MESSAGE field of a PZG_PEER_COMMAND_USER_MESSAGE Message will always be the Message that was passed in by subclass code, not a PZG-internal Message.

Reply:
You would think so, but not all messages get wrapped inside a PZG_PEER_NAME_USER_MESSAGE message.
Please see commit #5 where I also wrap the SendUnicastUserMessageToPeer case.